src/engine.ts

```diff
@@ -868,6 +882,18 @@ export default class Engine {
 
     pl.payCosts(cost);
     pl.gainRewards(income);
+
+    // check if it's a gaia phase
+    if (cost[0].type === Resource.GainTokenGaiaArea) {
+
+      if (pl.data.brainstone === BrainstoneArea.Transit) {
```

Comment:
But terrans can't have brainstone?

Reply:
Taklons have it, and they can gaiaform

Reply:
Except you're moving the brainstone in area 2 directly, as if the brainstone was owned by terrans.

This code is only reachable for terrans anyway, as it's a free action spending gaia tokens.

Reply:
Right!